### PR TITLE
camel workitems

### PR DESCRIPTION
**Pull request by tsurdilo**

### CODE REVIEW

**Comment by tsurdilo** [comment]
@mswiderski @krisv this ports the camel workitems.

File: camel-workitem/pom.xml [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>com.h2database</groupId>
+      <artifactId>h2</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.codehaus.btm</groupId>
+      <artifactId>btm</artifactId>
```
Comment (mswiderski) [inline camel-workitem/pom.xml]: we should not use bitronix any more in v7, this needs to be removed or replaced with narayana

File: pom.xml [hunk 2]
```diff
@@ -80,6 +81,7 @@
     <version.github.jnr>0.15</version.github.jnr>
     <version.squareup.okhttp3>3.8.1</version.squareup.okhttp3>
     <version.rxjava>1.2.4</version.rxjava>
+    <version.camel.redhat>2.17.0.redhat-630187</version.camel.redhat>
```
Comment (mswiderski) [inline pom.xml]: can we use this version instead 2.21.0 as this is our target (as far as I know)

File: pom.xml [hunk 3]
```diff
@@ -719,6 +742,21 @@
         <updatePolicy>daily</updatePolicy>
       </snapshots>
     </repository>
+
+    <repository>
+      <id>redhat-product-ga-repository</id>
+      <name>Red Hat Product Repository</name>
+      <url>https://maven.repository.redhat.com/ga/</url>
```
Comment (mswiderski) [inline pom.xml]: for community bits we should not rely on this repo

**Comment by tsurdilo** [comment]
@mswiderski updated, i had to use camel version 2.20.1 because with 2.21.0 the ftp tests were failing. Same code with 2.20.1 works fine. Idk what the difference is honestly.

**Comment by tsurdilo** [comment]
@mswiderski yes sorry I changed all the copyright years to 2018. Maybe its not too much noise

**Comment by tsurdilo** [comment]
Jenkins retest please

**Comment by tsurdilo** [comment]
@mswiderski @krisv is this ok to be merged?

**Comment by krisv** [comment]
@tsurdilo we probably shouldn't define the camel version in this repo, but rely on https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/master/pom.xml#L92
The version there probably also should be updated to camel v2.21.0, as this is the version that Fuse7 is using.  But maybe we can do a two-step approach?  As this is used in kie-camel so upgrade should be verified there as well.  Maybe change this PR to rely on the version defined in our bootstrap pom, and next try to upgrade the version in bootstrap repo?  The upgrade will be necessary but could potentially be done post-GA (if it would take more time to fix this FTP tests).

**Comment by tsurdilo** [comment]
@krisv amended - thanks. OK I will do a separate PR to update the boostrap repo to 2.21.0 and fix any issues with it.

**Pull request merged**